app/core/vectordb/chroma.py

```diff
@@ -59,9 +61,12 @@ def __init__(self, host=None, port=None, path="chromadb_store", collection_name=
             except Exception as exe:
                 raise ChromaException("While initializing client: "+str(exe)) from exe
         try:
+            # Check for passed embedding function, or use schema.EmbeddingType.DEFAULT
+            if not self.embedding_function:
+                embedding_function = SentenceTransformerEmbedding().get_embeddings
             self.db_conn = chroma_client.get_or_create_collection(
                 name=self.collection_name,
-                # embedding_function=custom_openai_emb_fn,
+                embedding_function=embedding_function,
                 )
```

Comment:
Is simply passing the new function like this working fine? Our function expects input in our own `document` format and all. Do we need any extra handling to get that working with chroma?

Reply:
Here I pass in the embedding function as the `EmbeddingInterface` (`SentenceTransformerEmbedding()` but then I just pass in just the following method:
```
def get_embeddings(self, doc_list: List[schema.Document]) -> None:
        '''Generates embeddings for the .text values and sets them to .embedding field of i/p items'''
        for doc in doc_list:
            doc.embedding = self.model.encode([doc.text.strip()])[0] 
```

It _seems_ like the embedding function is working, but I would have to confirm. If it's not embedding properly, I would just have to add a method that accepts an array of `Documents` as Chroma expects them to be, and then returns them with embeddings.
 
<img width="613" alt="Screenshot 2023-09-01 at 12 52 51 PM" src="https://github.com/BibleNLP/assistant.bible/assets/19649268/25f11aa1-04a4-476b-86fd-85dd2fa4709a">